apps/io-session-manager/src/controllers/authentication.ts

```diff
@@ -887,20 +938,15 @@ export const acs: (
 
 const isDifferentUserTryingToLogin = (
   spidUserFiscalCode: FiscalCode,
-  additionalProps?: AdditionalLoginPropsT,
-): boolean =>
-  pipe(
-    additionalProps?.currentUser,
-    O.fromNullable,
-    O.exists((c) => c !== sha256(spidUserFiscalCode)),
-  );
+  currentUserFiscalCode: string,
+): boolean => currentUserFiscalCode !== sha256(spidUserFiscalCode);
```

Comment:
We can introduce a type guard to avoid splitting the check between this function and the caller.
This way, the caller can safely access currentUserFiscalCodeOption.value within the guarded branch, keeping the logic cohesive and improving type safety.

```typescript
const isDifferentUserTryingToLogin = (
  spidUserFiscalCode: FiscalCode,
  currentUserFiscalCodeOption: O.Option<string>,
): currentUserFiscalCodeOption is O.Some<string> =>
  O.isSome(currentUserFiscalCodeOption) &&
  currentUserFiscalCodeOption.value !== sha256(spidUserFiscalCode);
```